### PR TITLE
[air/output] Use flat metrics in results report, use Trainable._progress_metrics

### LINKED ISSUE #34919: [AIR output] Too many metrics are reported by default for RL users
### What happened + What you expected to happen

By default, we report a long list of metrics for each iteration for RL users. It feels really spammy.
Can we choose fewer metrics for default output or report different sets of metrics for different algorithms
```
Training finished iter 13 at 2023-03-29 09:08:13 (running for 00:03:11.52)
agent_timesteps_total: 389000
connector_metrics: {}
counters:
  num_agent_steps_sampled: 389000
  num_agent_steps_trained: 388500
  num_env_steps_sampled: 389000
  num_env_steps_trained: 388500
  num_samples_added_to_queue: 389000
  num_training_step_calls_since_last_synch_worker_weights: 134
  num_weight_broadcasts: 964
custom_metrics: {}
episode_len_mean: 1721.88
episode_media: {}
episode_reward_max: 36.0
episode_reward_mean: 9.77
episode_reward_min: 4.0
episodes_this_iter: 67
episodes_total: 1710
info:
  learner:
    default_policy:
      custom_metrics: {}
      diff_num_grad_updates_vs_sampler_policy: 10.0
      learner_stats:
        cur_lr: 0.0005
        entropy: 1.0906739234924316
        entropy_coeff: 0.01
        policy_loss: -32.83232116699219
        total_loss: -28.335006713867188
        var_gnorm: 16.424108505249023
        vf_explained_var: 0.6170328259468079
        vf_loss: 19.683231353759766
      model: {}
      num_grad_updates_lifetime: 777.0
  learner_queue:
    size_count: 778
    size_mean: 0.0
    size_quantiles: [0.0, 0.0, 0.0, 0.0, 0.0]
    size_std: 0.0
  num_agent_steps_sampled: 389000
  num_agent_steps_trained: 388500
  num_env_steps_sampled: 389000
  num_env_steps_trained: 388500
  num_samples_added_to_queue: 389000
  num_training_step_calls_since_last_synch_worker_weights: 134
  num_weight_broadcasts: 964
  timing_breakdown:
    learner_dequeue_time_ms: 2772.957
    learner_grad_time_ms: 123.634
    learner_load_time_ms: 4.319
    learner_load_wait_time_ms: 47.829
num_agent_steps_sampled: 389000
num_agent_steps_trained: 388500
num_env_steps_sampled: 389000
num_env_steps_sampled_this_iter: 30750
num_env_steps_trained: 388500
num_env_steps_trained_this_iter: 31000
num_faulty_episodes: 0
num_healthy_workers: 10
num_in_flight_async_reqs: 20
num_remote_worker_restarts: 0
num_steps_trained_this_iter: 31000
perf:
  cpu_util_percent: 34.94117647058823
  ram_util_percent: 5.211764705882353
policy_reward_max: {}
policy_reward_mean: {}
policy_reward_min: {}
sampler_perf:
  mean_action_processing_ms: 0.6395067034460499
  mean_env_render_ms: 0.0
  mean_env_wait_ms: 7.840870172264184
  mean_inference_ms: 6.614064184668028
  mean_raw_obs_processing_ms: 2.9528597540097277
sampler_results:
  connector_metrics: {}
  custom_metrics: {}
  episode_len_mean: 1721.88
  episode_media: {}
  episode_reward_max: 36.0
  episode_reward_mean: 9.77
  episode_reward_min: 4.0
  episodes_this_iter: 67
  hist_stats:
    episode_lengths: [1414, 1306, 1641, 1446, 1234, 2026, 1600, 2454, 1359, 1572,
      1411, 1471, 1463, 1269, 1347, 1083, 2344, 1095, 1956, 1603, 1255, 2218, 1208,
      1943, 1483, 1158, 2108, 1073, 1535, 2590, 1804, 1802, 2109, 1783, 1099, 1258,
      1211, 1826, 2480, 1977, 1649, 1159, 1598, 1972, 2280, 2026, 1732, 1167, 1884,
      1599, 1722, 2156, 1723, 1767, 1387, 1849, 2061, 2356, 1875, 1727, 2524, 1620,
      1926, 1507, 1902, 1999, 1914, 1514, 1699, 1095, 2081, 1632, 1520, 1578, 2329,
      985, 1681, 1719, 1836, 1306, 2122, 1726, 1804, 2020, 2076, 1235, 1074, 1970,
      1853, 1836, 1228, 1431, 2112, 1946, 2793, 1822, 2044, 1946, 2200, 1880]
    episode_reward: [9.0, 12.0, 7.0, 7.0, 5.0, 11.0, 7.0, 17.0, 10.0, 11.0, 9.0, 6.0,
      6.0, 6.0, 9.0, 4.0, 13.0, 11.0, 12.0, 7.0, 5.0, 10.0, 5.0, 10.0, 14.0, 4.0,
      10.0, 4.0, 10.0, 15.0, 7.0, 11.0, 9.0, 8.0, 11.0, 5.0, 8.0, 16.0, 13.0, 8.0,
      6.0, 12.0, 6.0, 9.0, 11.0, 13.0, 7.0, 4.0, 11.0, 7.0, 10.0, 15.0, 7.0, 15.0,
      6.0, 8.0, 10.0, 36.0, 8.0, 8.0, 14.0, 9.0, 11.0, 13.0, 15.0, 9.0, 8.0, 5.0,
      8.0, 11.0, 14.0, 11.0, 6.0, 14.0, 20.0, 4.0, 11.0, 8.0, 8.0, 12.0, 14.0, 10.0,
      10.0, 11.0, 10.0, 4.0, 4.0, 9.0, 8.0, 8.0, 4.0, 5.0, 10.0, 11.0, 20.0, 13.0,
      14.0, 8.0, 13.0, 9.0]
  num_faulty_episodes: 0
  policy_reward_max: {}
  policy_reward_mean: {}
  policy_reward_min: {}
  sampler_perf:
    mean_action_processing_ms: 0.6395067034460499
    mean_env_render_ms: 0.0
    mean_env_wait_ms: 7.840870172264184
    mean_inference_ms: 6.614064184668028
    mean_raw_obs_processing_ms: 2.9528597540097277
time_this_iter_s: 11.578344583511353
time_total_s: 149.75555968284607
timers:
  sample_time_ms: 0.242
  synch_weights_time_ms: 0.027
  training_iteration_time_ms: 0.354
timesteps_total: 389000
training_iteration: 13
```

### Versions / Dependencies

nightly

### Reproduction script

n/a

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

We already have a `Trainable._progress_metrics` property that can be used to define default metrics to be displayed in the results table.

With this PR, we port this functionality to the new output. For rllib, an example output can look like this:

```
Trial PPO_CartPole-v1_cdf47_00000 finished iteration 4 at 2023-05-04 11:03:06 (running for 00:00:21.57).
╭─────────────────────────────────────────────────────╮
│ Trial PPO_CartPole-v1_cdf47_00000 result            │
├─────────────────────────────────────────────────────┤
│ episodes_total                                  423 │
│ num_env_steps_sampled                         28000 │
│ num_env_steps_trained                         28000 │
│ sampler_results/episode_len_mean             153.15 │
│ sampler_results/episode_reward_mean          153.15 │
╰─────────────────────────────────────────────────────╯
```

In order to make the implementation simpler, and also to increase readability, we move from the "short-flat" output to a full flattened output, where the full config path is displayed in the results and config tables. E.g.

```
...
model/use_lstm                                                      False │
│ model/vf_share_layers                                               False │
│ model/zero_mean                                                      True │
│ multiagent/count_steps_by                                       env_steps │
│ multiagent/observation_fn                                                 │
│ multiagent/policies/default_policy                   ...None, None, None) │
│ multiagent/policies_to_train                                              │
...
```

## Related issue number

Closes #34919

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm.py [hunk 1]
```diff
@@ -210,12 +210,14 @@ class Algorithm(Trainable):
...
-        "sampler_results/episode_reward_mean",
-        "evaluation/sampler_results/episode_reward_mean",
+    _progress_metrics = (
         "num_env_steps_sampled",
         "num_env_steps_trained",
-    ]
+        "episodes_total",
+        "episode_len_mean",
+        "episode_reward_mean",
+        "evaluation/episode_reward_mean",
```
Comment (sven1977) [inline rllib/algorithms/algorithm.py:219]: change last 3 items to:
```
sampler_results/episode_len_mean
sampler_results/episode_reward_mean
evaluation/sampler_results/episode_reward_mean
```

File: python/ray/tune/utils/callback.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def _create_default_callbacks(
     sync_config: SyncConfig,
     air_verbosity: Optional["AirVerbosity"] = None,
     metric: Optional[str] = None,
-    progress_metrics: Optional[List[str]] = None,
+    progress_metrics: Optional[Tuple] = None,
```
Comment (sven1977) [inline python/ray/tune/utils/callback.py:48]: Nit: There is no Args list in the docstring that allows me to check, what format e.g. `progress_metrics` should have and what the items in the Tuple should be like.

Comment (krfricke) [reply python/ray/tune/utils/callback.py]: Updated to `Collection[str]`!

File: rllib/algorithms/algorithm.py (hunk 1 above)
Comment (krfricke) [reply rllib/algorithms/algorithm.py]: Updated, PTAL!

**Review by sven1977 — APPROVED** [review]
LGTM! Thanks @krfricke

**Pull request merged**